MapleServer2/Managers/FieldManager.cs

```diff
@@ -724,7 +751,12 @@ public void AddPortal(IFieldObject<Portal> portal)
 
     public void RemovePortal(IFieldObject<Portal> portal)
     {
-        State.RemovePortal(portal.ObjectId);
+        if (portal is null)
```

Comment:
In what circumstances are we trying to remove a portal that is `null`?

Reply:
I can't say 100% for sure if this is it, but I vaguely recall this might have happened while configuring a portal in the housing. I see a code path that leads to it from a function called HandleChangePortalSettings and I remember there being some weirdness with some housing features in some scenarios.